src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.enums.Color;
+import core.basesyntax.enums.Figures;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANDOM_VALUE = 100;
+    private static final String DEFAULT_COLOR = Color.WHITE.name().toLowerCase();
+    private static final int DEFAULT_VALUE = 10;
+    private Random random;
+    private Figures figures;
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(Random random, ColorSupplier supplier) {
+        this.random = random;
+        this.colorSupplier = supplier;
+    }
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(Figures.values().length);
+        figures = Figures.values()[index];
+        String randomColor = colorSupplier.getRandomColor();
+        switch (figures) {
+            case CIRCLE:
+                return new Circle(colorSupplier.getRandomColor(), random.nextInt(MAX_RANDOM_VALUE));
```

Comment:
Let’s create a private method that returns the random parameter and avoid code duplication